openeo_processes_dask/process_implementations/cubes/merge.py

```diff
@@ -10,126 +8,159 @@
 
 __all__ = ["merge_cubes"]
 
+NEW_DIM_NAME = "cubes"
+NEW_DIM_COORDS = ["cube1", "cube2"]
+
+
+from collections import namedtuple
+
+Overlap = namedtuple("Overlap", ["only_in_cube1", "only_in_cube2", "in_both"])
+
 
 def merge_cubes(
     cube1: RasterCube,
     cube2: RasterCube,
     overlap_resolver: Callable = None,
     context: Optional[dict] = None,
     **kwargs,
-):
+) -> RasterCube:
+
     if context is None:
         context = {}
     if not isinstance(cube1, type(cube2)):
         raise Exception(
             f"Provided cubes have incompatible types. cube1: {type(cube1)}, cube2: {type(cube2)}"
         )
 
-    is_geopandas = isinstance(cube1, gpd.geodataframe.GeoDataFrame) and isinstance(
-        cube2, gpd.geodataframe.GeoDataFrame
-    )
-    is_delayed_geopandas = isinstance(
-        cube1, dask_geopandas.core.GeoDataFrame
-    ) and isinstance(cube2, dask_geopandas.core.GeoDataFrame)
-    if is_geopandas or is_delayed_geopandas:
-        if list(cube1.columns) == list(cube2.columns):
-            if is_delayed_geopandas:
-                merged_cube = cube1.append(cube2)
-            if is_geopandas:
-                merged_cube = cube1.append(cube2, ignore_index=True)
-            print(
-                "Warning - Overlap resolver is not implemented for geopandas vector-cubes, cubes are simply appended!"
+    # Key: dimension name
+    # Value: (labels in cube1 not in cube2, labels in cube2 not in cube1)
+    overlap_per_shared_dim = {
+        dim: Overlap(
+            only_in_cube1=np.setdiff1d(cube1[dim].data, cube2[dim].data),
+            only_in_cube2=np.setdiff1d(cube2[dim].data, cube1[dim].data),
+            in_both=np.intersect1d(cube1[dim].data, cube2[dim].data),
+        )
+        for dim in set(cube1.dims).intersection(set(cube2.dims))
+    }
+
+    differing_dims = set(cube1.dims).symmetric_difference(set(cube2.dims))
+
+    if len(differing_dims) == 0:
+        # Check whether all of the shared dims have exactly the same labels
+        dims_have_no_label_diff = all(
+            [
+                len(overlap.only_in_cube1) == 0 and len(overlap.only_in_cube2) == 0
+                for overlap in overlap_per_shared_dim.values()
+            ]
+        )
+        if dims_have_no_label_diff:
+            # Example 3: All dimensions and their labels are equal
+            concat_both_cubes = (
+                xr.concat([cube1, cube2], dim=NEW_DIM_NAME)
+                .reindex({NEW_DIM_NAME: NEW_DIM_COORDS})
+                .chunk({NEW_DIM_NAME: -1, "x": "auto", "y": "auto"})
```

Comment:
Can the chunk of the NEW_DIM_NAME: -1, which means that this dimension is not chunked, become too large for dask? (Maybe this is handled in "x": "auto", "y": "auto" automatically, but I am not sure? Also, do we always expect "x" and "y" to be there?

Reply:
Ah, no, NEW_DIM_NAME only ever has two coordinates, cube1 and cube2! Without this rechunking, the two cubes would live in different chunks, which doesn't work if we want to reduce over it!

Huh, actually good point on x and y. I included this, such that the config item of `array.chunk-size` is respected and chunks are broken down if too large - I'll have a look whether there's a way of doing this without specifying x and y explicitly!

Reply:
Updated!